File: archive/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.js
const isPrefixOfWord = function (sentence, searchWord) {
  sentence = sentence.split(' ')
  let ans = -1
  for (let i = 0; i < sentence.length; i++) {
    if (sentence[i].length < searchWord.length) {continue} else {
      let mem = ''
      for (let j = 0; j < searchWord.length; j++) {
        mem+=sentence[i][j]
      }

      if (mem === searchWord) {
        ans = i + 1
        break
      }
    }
  }
  return ans
};

console.log(isPrefixOfWord("this problem is an easy problem", "pro"))

File: check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.go
func isPrefixOfWord(sentence string, searchWord string) int {
    s:=strings.Split(sentence, " ")

    for i, v:= range s {
        if len(v) < len(searchWord) {continue}

        if searchWord == v[0:len(searchWord)] {return i + 1}
    }

    return -1
}